### PR TITLE
new tutorial: Migration guide from fastai's python library to FastAI.jl

### PR DESCRIPTION
**Pull request by War-Eagl**
First of three part tutorial on migrating from Fastai's python library to FastAI.jl. This is supposed to beginner level and I've written down things that I felt crucial when I came from the Fastai's python's library.

### CODE REVIEW

**Comment by darsnack** [comment]
I feel like one important part of fast.ai is the [data block API](https://docs.fast.ai/tutorial.datablock). This tutorial would benefit from a mapping from that API to our API (MLDataPattern.jl + DataAugmentation.jl).

File: docs/notebooks/Migration_Guide_beginner.ipynb [hunk 1]
```diff
@@ -0,0 +1,448 @@
...
+        "# DataLoaders"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "fFmwN26ZA5wO"
+      },
+      "source": [
+        "Dataloaders are important aspects of fastai's python library. These objects are like assembly lines that feed materials to the machine. Likewise dataloaders feed data to the model. \n",
```
Comment (darsnack) [inline docs/notebooks/Migration_Guide_beginner.ipynb:193]:
```suggestion
        "Dataloaders are important aspects of fastai's python library. Like an assembly line feeding materials to the next machine, dataloaders feed data to the model. \n",
```

File: docs/notebooks/Migration_Guide_beginner.ipynb [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "EmDKNgh2A3Dp"
+      },
+      "source": [
+        "# DataLoaders"
+      ]
+    },
```
Comment (darsnack) [inline docs/notebooks/Migration_Guide_beginner.ipynb:186]: My concern with this section is that it blurs the line between data containers and data iterators. For a migration guide, I would prefer to see an explanation of how in FastAI.jl, we have a single interface — `MLDataPattern.getobs` and `MLDataPattern.nobs` — for both data containers and data iterators. The only distinction is that data iterators like `eachbatch` or  `eachobs` also implement the [iteration interface](https://docs.julialang.org/en/v1/manual/interfaces/#man-interface-iteration) from Base.

This should be contrasted with fast.ai where there is a distinction between datasets and data loaders.

File: docs/notebooks/Migration_Guide_beginner.ipynb [hunk 3]
```diff
@@ -0,0 +1,448 @@
...
+        "# Learner"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "bdx1u49NFxjj"
+      },
+      "source": [
+        "Learner is the object that binds the data loaders with the model and sets the other hyper parameters like loss function, optimizers, etc. There is `Learner()` function which is similar to fastai's `cnn_learner` function. The below python code can be translated as,\n",
```
Comment (darsnack) [inline docs/notebooks/Migration_Guide_beginner.ipynb:377]:
```suggestion
        "The learner is the object that binds the data loaders with the model and sets the other hyper parameters like loss function, optimizers, etc. There is `Learner()` function which is similar to fastai's `cnn_learner` function. The below python code can be translated as,\n",
```

File: docs/notebooks/Migration_Guide_beginner.ipynb [hunk 4]
```diff
@@ -0,0 +1,448 @@
...
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "fHifey18IVqf"
+      },
+      "source": [
+        "However, `Learn` method makes us to define every parameters manually. Like we should explicitly convert our data container object to `dataloader` object and pass them. \n",
+        "\n",
+        "Instead we can cut the complexity by using `methodlearner` function where everything can be passed as hyperparameters. The cool thing about `methodlearner` is it takes the data container object and implicitly converts it into dataloader, where the batch size, validation split can be customized if we need. Therefore the above code can be converted into, "
+      ]
+    },
```
Comment (darsnack) [inline docs/notebooks/Migration_Guide_beginner.ipynb:419]: I would just delete this section. The comparison to `Learner` isn't exact, and I think `cnn_learner` is more closely compared to `methodlearner`.

**Review by darsnack — CHANGES_REQUESTED** [review]
Left some comments. Overall, I think this is a good start, but I would prefer to see more of a focus on "translating" concepts instead of code snippets.

**Comment by War-Eagl** [comment]
I corrected the typos and deleted those lines in Learner section. I also added some explanation in Dataloaders section. Does that make any sense  and do any justice to the tutorial?

**Pull request closed without merging**